app/assets/javascripts/training/components/training_slide_handler.jsx

```diff
@@ -59,13 +62,24 @@ const getSlideInfo = (training, locale) => {
 
 const keys = { rightKey: 39, leftKey: 37 };
 
+
+// helper variables for Alerthandler function.
```

Comment:
I don't like how much of this logic is getting added to the TrainingSlideHandler, especially considering that it's spread out ins two different sections. The handler file is already too complex.

Can this be done in a way where all or almost all of the logic for whether to display the modal is contained within the FastTrainingAlert file instead?

Reply:
Done! I was successful in moving the fastTrainingAlertFunction as well as all the helper variables to the FastTrainingAlert file. I have only called the function in TrainingSlideHandler and used one only more line for the Usestate hook as there is no other way around. Now, TrainingSlideHandler has barely three lines of code added.